test/k8sT/Policies.go

```diff
@@ -88,197 +80,421 @@ var _ = Describe("K8sPolicyTest", func() {
+
+			err = waitUntilEndpointUpdates(ciliumPod, eps, 4)
+			Expect(err).Should(BeNil())
+
+			By("Applying Policy in default namespace")
+			eps = kubectl.CiliumEndpointPolicyVersion(ciliumPod)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
+			err = waitUntilEndpointUpdates(ciliumPod, eps, 4)
+			Expect(err).Should(BeNil(), "Endpoints timeout on namespaces %q", helpers.DefaultNamespace)
+
+			By(fmt.Sprintf("Testing %s namespace", namespace))
+			clusterIP, _, err := kubectl.GetServiceHostPort(namespace, "app1-service")
+			Expect(err).To(BeNil(), "Cannot get service on %q namespace", namespace)
+			appPods := getAppPods(namespace)
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App2],
+				fmt.Sprintf("curl http://%s/public", clusterIP))
+			Expect(err).Should(BeNil())
+
+			_, err = kubectl.ExecPodCmd(
+				namespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(HaveOccurred())
+
+			By("Testing default namespace")
+			clusterIP, _, err = kubectl.GetServiceHostPort(helpers.DefaultNamespace, "app1-service")
+			Expect(err).To(BeNil(), "Cannot get service on default namespace")
+
+			appPods = getAppPods(helpers.DefaultNamespace)
+			_, err = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				fmt.Sprintf("curl http://%s/public", clusterIP))
+			Expect(err).Should(BeNil())
+
+			_, err = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			Expect(err).Should(HaveOccurred())
+		})
+	})
+
+	Context("GuestBook Examples", func() {
+		var (
+			deployment                = "guestbook_deployment.json"
+			groupLabel                = "zgroup=guestbook"
+			redisPolicy               = "guestbook-policy-redis.json"
+			redisPolicyName           = "guestbook-redis"
+			redisPolicyDeprecated     = "guestbook-policy-redis-deprecated.json"
+			redisPolicyDeprecatedName = "guestbook-redis-deprecated"
+			webPolicy                 = "guestbook-policy-web.yaml"
+			webPolicyName             = "guestbook-web"
+		)
+
+		var ciliumPod, ciliumPod2 string
+		var err error
+
+		BeforeEach(func() {
+			kubectl.Apply(kubectl.ManifestGet(deployment))
+
+			ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil())
+
+			ciliumPod2, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
+			Expect(err).Should(BeNil())
+		})
+
+		AfterEach(func() {
+			kubectl.Delete(kubectl.ManifestGet(deployment))
+			kubectl.Delete(kubectl.ManifestGet(webPolicy))
+			kubectl.Delete(kubectl.ManifestGet(redisPolicy))
+			kubectl.Delete(kubectl.ManifestGet(redisPolicyDeprecated))
```

Comment:
Since calls to delete resources in Kubernetes are asynchronous, add checks to ensure that deletion occurred successfully, as we want to ensure that the test only ends when everything is cleaned up.